File: src/main/java/com/example/storage/controller/dto/IssueDto.java
package com.example.storage.controller.dto;

import com.example.storage.entity.Issue;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class IssueDto {
    private long id;
    private EquipmentDto equipment;
    private LocalDate issueDate;
    private LocalDate returnDate;

    public static IssueDto toDto(Issue issue) {
        return new IssueDto(
                issue.getId(),
                EquipmentDto.toDto(issue.getEquipment()),
                issue.getIssueDate(),
                issue.getReturnDate()
        );
    }
}

File: src/main/kotlin/ru/nto/storage/controller/dto/IssueDto.kt
package ru.nto.storage.controller.dto

import ru.nto.storage.entity.Issue
import java.time.LocalDate

data class IssueDto(
    val id: Long,
    val equipment: EquipmentDto,
    val issueDate: LocalDate,
    val returnDate: LocalDate
) {
    companion object {
        fun toDto(issue: Issue) = IssueDto(
            id = issue.id,
            equipment = EquipmentDto.toDto(issue.equipment),
            issueDate = issue.issueDate,
            returnDate = issue.returnDate
        )
    }
}
